### PR TITLE
api-docs: add wallet http auction endpoints

### PR DESCRIPTION
**Pull request by pinheadmz**
Split off from https://github.com/handshake-org/handshake-org.github.io/pull/41

### CODE REVIEW

File: src/api-docs/source/includes/_wallet_auctions.md [hunk 1]
```diff
@@ -1,179 +1,912 @@
...
+List the states of all names known to the wallet.
+
+### HTTP Request
+
+`GET /wallet/:id/name`
+
+
+Parameters | Description
+---------- | -----------
+id <br> _string_ | id of wallet you would like to retrieve the name list for
```
Comment (boymanjor) [inline src/api-docs/source/includes/_wallet_auctions.md]: suggestion: these wallet descriptions are similar to the name descriptions. We could simplify them to "wallet id."

File: src/api-docs/source/includes/_wallet_auctions.md [hunk 2]
```diff
@@ -1,179 +1,912 @@
...
+      "index": 0
+    },
+    "value": 10000000,
+    "height": 221,
+    "own": false
+  }
+]
+```
+
+List all bids for all names known to the wallet.
```
Comment (boymanjor) [inline src/api-docs/source/includes/_wallet_auctions.md]: s/bids/reveals/

File: src/api-docs/source/includes/_wallet_auctions.md [hunk 3]
```diff
@@ -1,179 +1,912 @@
...
+    },
+    "value": 10000000,
+    "height": 221,
+    "own": false
+  }
 ]
 ```
 
-Your wallet tracks any name on which you have bid or opened. `getnames` returns info on each. This is different from the `hsd-rpc getnames` call which returns info on all names for which the node has data.
+List the bids for a specific name known to the wallet.
```
Comment (boymanjor) [inline src/api-docs/source/includes/_wallet_auctions.md]: s/bids/reveals/

File: src/api-docs/source/includes/_wallet_auctions.md [hunk 4]
```diff
@@ -1,179 +1,912 @@
...
+      "type": "TXT",
+      "txt": [
+        "A decentralized DNS root zone!"
+      ]
+    }
+  ]
+}
+```
+
+List all bids for all names known to the wallet.
```
Comment (boymanjor) [inline src/api-docs/source/includes/_wallet_auctions.md]: This description is wrong.

File: src/api-docs/source/includes/_wallet_auctions.md [hunk 5]
```diff
@@ -1,179 +1,912 @@
...
+
+### HTTP Request
+
+`GET /wallet/:id/reveal`
```
Comment (boymanjor) [inline src/api-docs/source/includes/_wallet_auctions.md]: This endpoint and following description is wrong too.

File: src/api-docs/source/includes/_wallet_auctions.md [hunk 6]
```diff
@@ -329,407 +1033,409 @@ const client = new NodeClient(clientOptions);
...
 ```
-The BIDDING period is followed by the REVEAL period, during which bidders must reveal their bids, moving the inputs of their bids into a REVEAL covenant from which they must continue to REDEEM (loser) or REGISTER (winner).
 
-### Params
-Name | Default |  Description
---------- | --------- | ---------
-name | Required | The name for which you wish to reveal your bid
+Create, sign, and send a name REVEAL. If multiple bids were placed on a name,
+all bids will be revealed by this transaction. If no value is passed in for `name`,
+all bids for all names in the wallet will be sent.
```
Comment (boymanjor) [inline src/api-docs/source/includes/_wallet_auctions.md]: s/bids/reveals/

**Comment by pinheadmz** [comment]
@boymanjor addressed all nits and did a big clean up of the wallet rpc stuff, simplifying all the descriptions and parameters, fixing lots of mistakes in the examples. Also rebased this PR so it doesnt conflict with the other PRs.

**Pull request merged**